base/file_sys.h

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#ifdef _WIN32
+#include <Windows.h>
+#endif  // _WIN32
+
+#include <sys/stat.h>
+
+#include <string>
+#include <fstream>
+
+namespace ort_extensions {
+
+class path {
+ public:
+  path() = default;
+  path(const std::string& path) : path_(path){};
```

Comment:
On Windows it should be wstring. Otherwise you could hit path length size limit

Reply:
Yes, it was done differently. the input string here is UTF-8 and on Windows it will be converted to UTF-16 before any file access API.